java/food-packaging/src/main/java/org/acme/foodpackaging/domain/JobOperatorUpdatingVariableListener.java

```diff
@@ -0,0 +1,50 @@
+package org.acme.foodpackaging.domain;
+
+import ai.timefold.solver.core.api.domain.variable.VariableListener;
+import ai.timefold.solver.core.api.score.director.ScoreDirector;
+import org.jspecify.annotations.NonNull;
+
+import java.util.Objects;
+
+public class JobOperatorUpdatingVariableListener implements VariableListener<PackagingSchedule, Job> {
+
+    private static final String LINE_OPERATOR_FIELD = "lineOperator";
+
+    @Override
+    public void beforeVariableChanged(@NonNull ScoreDirector<PackagingSchedule> scoreDirector, @NonNull Job job) {
+        // Empty method
+    }
+
+    @Override
+    public void afterVariableChanged(@NonNull ScoreDirector<PackagingSchedule> scoreDirector, @NonNull Job job) {
+        if (job.getLine() == null && job.getLineOperator() != null) {
```

Comment:
Don't repeat yourself so often. Extract `job.getLineOperator()` and `job.getLine()` to their own local vars. Will make the code easier to read.